src/js/controller/controller.js

```diff
@@ -471,6 +474,7 @@ Object.assign(Controller.prototype, {
             if (adState) {
                 if (model.get('autoPause').pauseAds) {
                     _pauseWhenNotViewable(viewable);
+                    _this._instreamAdapter.noResume = !viewable;
                 } else {
                     _pauseAfterAd(viewable);
                 }
```

Comment:
Can't we also just use `_pauseAfterAd` here?

```suggestion
                    _pauseWhenNotViewable(viewable);
                }
                _pauseAfterAd(viewable);
```

Reply:
```
        function _pauseAfterAd(viewable) {
            _this._instreamAdapter.noResume = !viewable;
            if (!viewable) {
                _updatePauseReason({ reason: 'viewable' });
            }
        }
```